pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseSingleStageBrokerRequestHandler.java

```diff
@@ -366,18 +368,22 @@ protected BrokerResponse handleRequest(long requestId, String query, @Nullable S
       BrokerRequest brokerRequest = CalciteSqlCompiler.convertToBrokerRequest(pinotQuery);
       BrokerRequest serverBrokerRequest =
           serverPinotQuery == pinotQuery ? brokerRequest : CalciteSqlCompiler.convertToBrokerRequest(serverPinotQuery);
-      boolean hasTableAccess =
-          accessControl.hasAccess(requesterIdentity, serverBrokerRequest) && accessControl.hasAccess(httpHeaders,
-              TargetType.TABLE, tableName, Actions.Table.QUERY);
+      AuthorizationResult authorizationResult = accessControl.authorize(requesterIdentity, serverBrokerRequest);
+      if (authorizationResult.hasAccess()) {
+        authorizationResult = BasicAuthorizationResultImpl.joinResults(authorizationResult,
```

Comment:
We don't need to join results since the first auth already pass. Joining multiple auth result seems an uncommon operation